components/cam/src/control/interpolate_data.F90

```diff
@@ -977,6 +977,96 @@ subroutine vertinterp(ncol, ncold, nlev, pmid, pout, arrin, arrout)
     return
   end subroutine vertinterp
 
+  subroutine vertinterpz(ncol, ncold, nlev, zmid, zout, arrin, arrout)
+
+    !-----------------------------------------------------------------------
+    !
+    ! Purpose: Vertically interpolate input array to output height (above 
+    !          surface) level.  Copy values at boundaries.
+    !
+    ! Method:  Copies vertinterp and shifts its logic so height decreases 
+    !          with increasing level index.
+    !
+    ! Author:  Bryce Harrop
+    ! Note:    A more elegant method would be to introduce controlling logic
+    !          to vertinterp to get rid of the need for two separate 
+    !          subroutines.
+    !          Simple tests were done for linear and logarithmic 
+    !          interpolation, and the results suggested linear is better.
```

Comment:
Suggest updating this comment to be more precise: "linear is better for interpolation of low-level winds"